src/utils.ts

```diff
@@ -0,0 +1,61 @@
+import { Request } from 'express';
+import fs from 'fs/promises'
+import { Cart, Product } from './types';
+
+export const getUserIdFromHeaders = (req: Request): string | null => {
+    const userId = req.headers['x-user-id'] as string;
+
+    if (userId) {
+        return userId;
+    }
+
+    return null;
+}
```

Comment:
I would rewrite it a bit:

```suggestion
export const getUserIdFromHeaders = (
  req: Request
): string | string[] | undefined => {
  return req.headers["x-user-id"];
};
```